msal/src/Microsoft.Identity.Client/Telemetry.cs

```diff
@@ -201,11 +207,36 @@ private List<EventBase> CollateOrphanedEvents(String requestId)
                 {
                     // The orphaned event already contains its own start time, we simply collect it
                     EventBase orphan;
-                    EventsInProgress.TryRemove(key, out orphan);
+                    EventsInProgress.TryRemove(key, out orphan);
+
+                    IncrementEventCount(requestId, orphan);
                     orphanedEvents.Add(orphan);
                 }
             }
             return orphanedEvents;
+        }
+
+        private void IncrementEventCount(string requestId, EventBase eventToIncrement)
+        {
+            string eventName;
+            if (eventToIncrement[EventBase.EventNameKey].Substring(0,10) == "msal.token")

```

Comment:
Are you sure the eventToIncrement has that key? You should probably check first to avoid a null ref.

Reply:
eventToIncrement will always have that key, it is a requirement for instantiation of EventBase